Can't use AOT with config URLs that contain `$`

### Expected Behavior

Expect that any legal YAML config that is usable with Micronaut also to be usable with AOT

I have the following URL intercept map:
```yaml
micronaut:
  security:
    intercept-url-map:
      # (other URLs omitted for brevity)
      -
        pattern: /v1/odata/$metadata
        http-method: GET
```

When I run the regular non-AOT build, it works fine and handles the URL with the dollar sign present.

### Actual Behaviour

When I run the regular non-AOT build, it works fine and handles the URL with the dollar sign present. With the AOT build active and YAML->Java compilation turned off, things also work fine. With YAML->Java enabled, I get:

```
Caused by: java.lang.IllegalArgumentException: index 1 for '$m' not in range (received 0 arguments)
        at com.squareup.javapoet.Util.checkArgument(Util.java:53)
        at com.squareup.javapoet.CodeBlock$Builder.add(CodeBlock.java:287)
        at com.squareup.javapoet.CodeBlock$Builder.addStatement(CodeBlock.java:402)
        at com.squareup.javapoet.MethodSpec$Builder.addStatement(MethodSpec.java:525)
        at io.micronaut.aot.core.codegen.MapGenerator.generateMapMethod(MapGenerator.java:121)
        at io.micronaut.aot.core.codegen.MapGenerator.convertValueToSource(MapGenerator.java:63)
        at io.micronaut.aot.core.codegen.MapGenerator.generateListMethod(MapGenerator.java:100)
        at io.micronaut.aot.core.codegen.MapGenerator.convertValueToSource(MapGenerator.java:61)
        at io.micronaut.aot.core.codegen.MapGenerator.generateMap(MapGenerator.java:44)
        at io.micronaut.aot.std.sourcegen.MapPropertySourceGenerator.generate(MapPropertySourceGenerator.java:77)
        at io.micronaut.aot.core.codegen.AbstractSingleClassFileGenerator.generate(AbstractSingleClassFileGenerator.java:35)
        at io.micronaut.aot.std.sourcegen.YamlPropertySourceGenerator.createMapProperty(YamlPropertySourceGenerator.java:75)
        at io.micronaut.aot.std.sourcegen.YamlPropertySourceGenerator.generate(YamlPropertySourceGenerator.java:58)
        at io.micronaut.aot.std.sourcegen.AbstractStaticServiceLoaderSourceGenerator.generate(AbstractStaticServiceLoaderSourceGenerator.java:106)
        at io.micronaut.aot.core.codegen.ApplicationContextConfigurerGenerator.generate(ApplicationContextConfigurerGenerator.java:66)
        at io.micronaut.aot.MicronautAotOptimizer$Runner.execute(MicronautAotOptimizer.java:343)
        at io.micronaut.aot.MicronautAotOptimizer.execute(MicronautAotOptimizer.java:202)
        ... 15 more
```

It appears to be evaluating this `$` as a format character. The following mitigations appeared to do nothing:
- Single-quoting the string
- Double-quoting the string
- Escaping (single `\`) before the `$`

### Steps To Reproduce

1. With Micronaut AOT and YAML->Java enabled...
2. Add the above intercept URL config, or just a URL containing a `$`
3. Run the build

### Environment Information

- Operating system: macOS Ventura
- JDK Version: 17

### Example Application

_No response_

### Version

3.7.3

--- a/aot-core/src/main/java/io/micronaut/aot/core/codegen/MapGenerator.java
+++ b/aot-core/src/main/java/io/micronaut/aot/core/codegen/MapGenerator.java
@@ -118,7 +118,7 @@ public class MapGenerator {
         } else {
             mapMethod.addStatement("$T result = new $T<>($L)", Map.class, LinkedHashMap.class, value.size());
             for (Map.Entry<?, ?> entry : value.entrySet()) {
-                mapMethod.addStatement("result.put(" + convertValueToSource(entry.getKey(), builder) + ", " + convertValueToSource(entry.getValue(), builder) + ")");
+                mapMethod.addCode("result.put($L, $L);\n", convertValueToSource(entry.getKey(), builder), convertValueToSource(entry.getValue(), builder));
             }
             mapMethod.addStatement("return result");
         }

--- a/aot-std-optimizers/src/test/groovy/io/micronaut/aot/std/sourcegen/YamlPropertySourceGeneratorTest.groovy
+++ b/aot-std-optimizers/src/test/groovy/io/micronaut/aot/std/sourcegen/YamlPropertySourceGeneratorTest.groovy
@@ -55,7 +55,7 @@ public class Test_configStaticPropertySource extends MapPropertySource {
         put("micronaut.server.port", 8181);
         put("micronaut.server.cors.enabled", true);
         put("micronaut.security.intercept-url-map", list0());
-        put("otel.exclusions", list5());
+        put("otel.exclusions", list6());
         }});
   }
 
@@ -98,7 +98,7 @@ public class Test_configStaticPropertySource extends MapPropertySource {
     return result;
   }
 
-  private static List list5() {
+  private static List list6() {
     List result = new ArrayList<>(3);
     result.add("\${my.prop1}");
     result.add("\${my.prop2}");

--- a/aot-std-optimizers/src/test/groovy/io/micronaut/aot/std/sourcegen/YamlPropertySourceGeneratorTest.groovy
+++ b/aot-std-optimizers/src/test/groovy/io/micronaut/aot/std/sourcegen/YamlPropertySourceGeneratorTest.groovy
@@ -83,10 +83,18 @@ public class Test_configStaticPropertySource extends MapPropertySource {
     return result;
   }
 
+  private static Map map5() {
+    Map result = new LinkedHashMap<>(2);
+    result.put("pattern", "/v1/odata/\$metadata");
+    result.put("http-method", "GET");
+    return result;
+  }
+
   private static List list0() {
-    List result = new ArrayList<>(2);
+    List result = new ArrayList<>(3);
     result.add(map1());
     result.add(map3());
+    result.add(map5());
     return result;
   }
 

--- a/aot-std-optimizers/src/test/resources/test-config.yml
+++ b/aot-std-optimizers/src/test/resources/test-config.yml
@@ -21,6 +21,9 @@ micronaut:
         http-method: GET
         access:
           - isAnonymous()
+      -
+        pattern: /v1/odata/$metadata
+        http-method: GET
 otel:
   exclusions:
     - ${my.prop1}

